src/audio/mixin_mixout/mixin_mixout.c

```diff
@@ -707,16 +717,17 @@ static int mixout_params(struct processing_module *mod)
 		return -EINVAL;
 	}
 
-	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
+	/* FIXME: the code below could/should be removed once mess with having both frame_fmt
+	 * and valid_sample_fmt is solved (see comment in mixin_params()).
+	 */
 
 	/* comp_verify_params() does not modify valid_sample_fmt (a BUG?), let's do this here */
 	audio_stream_fmt_conversion(mod->priv.cfg.base_cfg.audio_fmt.depth,
 				    mod->priv.cfg.base_cfg.audio_fmt.valid_bit_depth,
 				    &frame_fmt, &valid_fmt,
 				    mod->priv.cfg.base_cfg.audio_fmt.s_type);
 
-	audio_stream_set_valid_fmt(&sink->stream, valid_fmt);
-	audio_stream_set_channels(&sink->stream, params->channels);
```

Comment:
Just in case, for eagle-eye reviewers :) , the removal of line with audio_stream_set_channels() is intentional. Channels number is already set above by comp_verify_params(), no need to set it again.